mlflow-integration UAT fails due to scikit-learn dependency version

test is failing with `AttributeError: module 'sklearn.metrics' has no attribute 'SCORERS'`
the current version of sklearn in requirements.txt is `scikit-learn==1.3.0` which is incompatible with the mlflow client

<details><summary>Full Logs</summary>

```bash
______________________ test_notebook[mlflow-integration] _______________________

test_notebook = '/tests/notebooks/mlflow/mlflow-integration.ipynb'

    @pytest.mark.ipynb
    @pytest.mark.parametrize(
        # notebook - ipynb file to execute
        "test_notebook",
        NOTEBOOKS.values(),
        ids=NOTEBOOKS.keys(),
    )
    def test_notebook(test_notebook):
        """Test Notebook Generic Wrapper."""
        os.chdir(os.path.dirname(test_notebook))
    
        with open(test_notebook) as nb:
            notebook = nbformat.read(nb, as_version=nbformat.NO_CONVERT)
    
        ep = ExecutePreprocessor(
            timeout=-1, kernel_name="python3", on_notebook_start=install_python_requirements
        )
        ep.skip_cells_with_tag = "pytest-skip"
    
        try:
            log.info(f"Running {os.path.basename(test_notebook)}...")
>           output_notebook, _ = ep.preprocess(notebook, {"metadata": {"path": "./"}})

/tests/test_notebooks.py:45: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/opt/conda/lib/python3.8/site-packages/nbconvert/preprocessors/execute.py:100: in preprocess
    self.preprocess_cell(cell, resources, index)
/opt/conda/lib/python3.8/site-packages/nbconvert/preprocessors/execute.py:121: in preprocess_cell
    cell = self.execute_cell(cell, index, store_history=True)
/opt/conda/lib/python3.8/site-packages/jupyter_core/utils/__init__.py:166: in wrapped
    return loop.run_until_complete(inner)
/opt/conda/lib/python3.8/asyncio/base_events.py:616: in run_until_complete
    return future.result()
/opt/conda/lib/python3.8/site-packages/nbclient/client.py:1021: in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <nbconvert.preprocessors.execute.ExecutePreprocessor object at 0x7ff4bf4fcb20>
cell = {'cell_type': 'code', 'execution_count': 10, 'id': 'a4295ea1-93fb-4100-acac-9191010caddc', 'metadata': {'execution': {...], 'source': '# run experiments\nruns = [\n    experiment(0.5, 0.5),\n    experiment(1, 0),\n    experiment(0, 1),\n]'}
cell_index = 11
exec_reply = {'buffers': [], 'content': {'ename': 'AttributeError', 'engine_info': {'engine_id': -1, 'engine_uuid': '78312f93-9666-...e, 'engine': '78312f93-9666-4d6a-a5fb-eff9b8862ff4', 'started': '2023-10-02T10:45:46.205665Z', 'status': 'error'}, ...}

    async def _check_raise_for_error(
        self, cell: NotebookNode, cell_index: int, exec_reply: t.Optional[t.Dict]
    ) -> None:
    
        if exec_reply is None:
            return None
    
        exec_reply_content = exec_reply['content']
        if exec_reply_content['status'] != 'error':
            return None
    
        cell_allows_errors = (not self.force_raise_errors) and (
            self.allow_errors
            or exec_reply_content.get('ename') in self.allow_error_names
            or "raises-exception" in cell.metadata.get("tags", [])
        )
        await run_hook(
            self.on_cell_error, cell=cell, cell_index=cell_index, execute_reply=exec_reply
        )
        if not cell_allows_errors:
>           raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
E           nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
E           ------------------
E           # run experiments
E           runs = [
E               experiment(0.5, 0.5),
E               experiment(1, 0),
E               experiment(0, 1),
E           ]
E           ------------------
E           
E           ---------------------------------------------------------------------------
E           AttributeError                            Traceback (most recent call last)
E           Cell In[10], line 3
E                 1 # run experiments
E                 2 runs = [
E           ----> 3     experiment(0.5, 0.5),
E                 4     experiment(1, 0),
E                 5     experiment(0, 1),
E                 6 ]
E           
E           Cell In[9], line 2, in experiment(alpha, l1_ratio)
E                 1 def experiment(alpha, l1_ratio):
E           ----> 2     mlflow.sklearn.autolog()
E                 3     with mlflow.start_run(run_name='wine_models', experiment_id=experiment_id) as run:
E                 4             mlflow.set_tag("author", "kf-testing")
E           
E           File /opt/conda/lib/python3.8/site-packages/mlflow/utils/autologging_utils/__init__.py:414, in autologging_integration.<locals>.wrapper.<locals>.autolog(*args, **kwargs)
E               395 with set_mlflow_events_and_warnings_behavior_globally(
E               396     # MLflow warnings emitted during autologging setup / enablement are likely
E               397     # actionable and relevant to the user, so they should be emitted as normal
E              (...)
E               410     disable_warnings=is_silent_mode,
E               411 ):
E               412     _check_and_log_warning_for_unsupported_package_versions(name)
E           --> 414     return _autolog(*args, **kwargs)
E           
E           File /opt/conda/lib/python3.8/site-packages/mlflow/sklearn/__init__.py:1214, in autolog(log_input_examples, log_model_signatures, log_models, disable, exclusive, disable_for_unsupported_versions, silent, max_tuning_runs, log_post_training_metrics, serialization_format, registered_model_name, pos_label)
E               936 @autologging_integration(FLAVOR_NAME)
E               937 def autolog(
E               938     log_input_examples=False,
E              (...)
E               949     pos_label=None,
E               950 ):  # pylint: disable=unused-argument
E               951     """
E               952     Enables (or disables) and configures autologging for scikit-learn estimators.
E               953 
E              (...)
E              1212                       be logged. If used for regression model, the parameter will be ignored.
E              1213     """
E           -> 1214     _autolog(
E              1215         flavor_name=FLAVOR_NAME,
E              1216         log_input_examples=log_input_examples,
E              1217         log_model_signatures=log_model_signatures,
E              1218         log_models=log_models,
E              1219         disable=disable,
E              1220         exclusive=exclusive,
E              1221         disable_for_unsupported_versions=disable_for_unsupported_versions,
E              1222         silent=silent,
E              1223         max_tuning_runs=max_tuning_runs,
E              1224         log_post_training_metrics=log_post_training_metrics,
E              1225         serialization_format=serialization_format,
E              1226         pos_label=pos_label,
E              1227     )
E           
E           File /opt/conda/lib/python3.8/site-packages/mlflow/sklearn/__init__.py:1767, in _autolog(flavor_name, log_input_examples, log_model_signatures, log_models, disable, exclusive, disable_for_unsupported_versions, silent, max_tuning_runs, log_post_training_metrics, serialization_format, pos_label)
E              1762     for metric_name in _get_metric_name_list():
E              1763         safe_patch(
E              1764             flavor_name, sklearn.metrics, metric_name, patched_metric_api, manage_run=False
E              1765         )
E           -> 1767     for scorer in sklearn.metrics.SCORERS.values():
E              1768         safe_patch(flavor_name, scorer, "_score_func", patched_metric_api, manage_run=False)
E              1770 def patched_fn_with_autolog_disabled(original, *args, **kwargs):
E           
E           AttributeError: module 'sklearn.metrics' has no attribute 'SCORERS'
E           AttributeError: module 'sklearn.metrics' has no attribute 'SCORERS'

/opt/conda/lib/python3.8/site-packages/nbclient/client.py:915: CellExecutionError

During handling of the above exception, another exception occurred:

test_notebook = '/tests/notebooks/mlflow/mlflow-integration.ipynb'

    @pytest.mark.ipynb
    @pytest.mark.parametrize(
        # notebook - ipynb file to execute
        "test_notebook",
        NOTEBOOKS.values(),
        ids=NOTEBOOKS.keys(),
    )
    def test_notebook(test_notebook):
        """Test Notebook Generic Wrapper."""
        os.chdir(os.path.dirname(test_notebook))
    
        with open(test_notebook) as nb:
            notebook = nbformat.read(nb, as_version=nbformat.NO_CONVERT)
    
        ep = ExecutePreprocessor(
            timeout=-1, kernel_name="python3", on_notebook_start=install_python_requirements
        )
        ep.skip_cells_with_tag = "pytest-skip"
    
        try:
            log.info(f"Running {os.path.basename(test_notebook)}...")
            output_notebook, _ = ep.preprocess(notebook, {"metadata": {"path": "./"}})
            # persist the notebook output to the original file for debugging purposes
            save_notebook(output_notebook, test_notebook)
        except CellExecutionError as e:
            # handle underlying error
>           pytest.fail(f"Notebook execution failed with {e.ename}: {e.evalue}")
E           Failed: Notebook execution failed with AttributeError: module 'sklearn.metrics' has no attribute 'SCORERS'
```
</details>

--- a/tests/notebooks/mlflow/requirements.in
+++ b/tests/notebooks/mlflow/requirements.in
@@ -3,4 +3,6 @@ boto3
 minio<7.0
 # pin the client to match the version of the deployed MLflow server
 mlflow==2.1.1
+# pin to ensure compatibility with the installed mlflow client
+scikit-learn<1.2
 tenacity

--- a/tests/notebooks/mlflow/requirements.txt
+++ b/tests/notebooks/mlflow/requirements.txt
@@ -154,8 +154,9 @@ requests==2.31.0
     #   mlflow
 s3transfer==0.6.2
     # via boto3
-scikit-learn==1.3.0
+scikit-learn==1.1.3
     # via
+    #   -r requirements.in
     #   mlflow
     #   shap
 scipy==1.10.1

